frontend/components/songs/song_index.jsx

```diff
@@ -1,16 +1,37 @@
 import React from "react";
+import { useState, useEffect } from "react";
 import SongCard from "./song_card";
+import SongCardDropdown from "./song_card_dropdown";
 
 
 const SongIndex = ({
     source,
     songs,
     history,
-    songCardDropdownState,
-    openSongCardDropdown,
-    closeSongCardDropdown,
+    params,
 }) => {
 
+    const [songCardDropdownState, setSongCardDropdownState] = useState({ isOpen: false });
+    const [selectedSongId, setSelectedSongId] = useState(null);
+    const [playlistedId, setPlaylistedId] = useState(null);
+    
+    const updateSongCardDropdownState = (newState) => {
+        setSongCardDropdownState(newState);
+    };
+    const updateselectedSongId = (newState) => {
+        setSelectedSongId(newState);
+    };
+    const updatePlaylistedId = (newState) => {
+        setPlaylistedId(newState);
+    };
+    
+    useEffect( () => {
+        return () => {
+            updateselectedSongId(null);
+            updateSongCardDropdownState({ isOpen: false });
+        }
+    }, [params]);
```

Comment:
What is params? Can we be more specific in the variable naming? It’s unclear to me when this effect will re-run.

Reply:
I understand; it is referring to URL parameters. I have deferred renaming this until the end since throughout the entire project, params consistently refers to URL params.